File: python_leaoai/python_leaoai/models/PainelConfig/Menu.py
# models/menu.py

from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship, Mapped, mapped_column
from database import Base
from .page import Page  # assumindo que você terá o modelo Page

class Menu(Base):
    __tablename__ = "menu"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    nome: Mapped[str] = mapped_column(String, nullable=True)
    role_gate: Mapped[str] = mapped_column(String, nullable=True)
    icon: Mapped[str] = mapped_column(String, nullable=True)

    # Relação com Page
    page: Mapped[list[Page]] = relationship("Page", back_populates="menu", cascade="all, delete-orphan")

File: Framework.Data/Models/PainelConfig/Menu.cs
using System.Collections.Generic;

namespace Data.Models.PainelConfig
{
    public class Menu : Base
    {
        public Menu()
        {
            this.Page = new HashSet<Page>();
        }

        public virtual ICollection<Page> Page { get; set; }

        public string Nome { get; set; }
        public string RoleGate { get; set; }
        public string Icon { get; set; }

    }
}
